### ISSUE TITLE
prank call check wrong sender balance

### ISSUE THREAD

**Issue by algtm** [ethereum/hevm#343]
same as https://github.com/ethereum/hevm/pull/245 
check wrong sender balance after prank
reproduce
echidna 2.2.1 hevm 0.51.3
```
let this.balance=0
let msg.sender.balance=100
let to = 0x10000
hevm.prank(msg.sender)
address(to).call{value:1}("")  //faild with BalanceTooLow 0x1,0x0
```

**Comment by d-xo** [comment]
Thanks for the report!

Can you provide a simple solidity file that I can use to reproduce?

**Comment by algtm** [comment]
@d-xo 
demo.sol
```
interface IHevm {
     function prank(address newSender) external;
}
contract BalanceDemo {
    address constant HEVM_ADDRESS = 0x7109709ECfa91a80626fF3989D68f67F5b1DD12D;
    IHevm hevm = IHevm(HEVM_ADDRESS);
    event LogUintAddr(string functionName, uint256 value, address addr);

    function demo()public{
        //send this.balance to msg.sender
        address(msg.sender).call{value: address(this).balance}("");
        emit LogUintAddr("this.balance", address(this).balance, address(this));
        emit LogUintAddr("sender.balance", address(msg.sender).balance, address(msg.sender));
        hevm.prank(msg.sender);
        address(0).call{value: 1}("");
        assert(false);
    }

}
```
config.yaml
```
testMode: assertion
```
echidna ./demo.sol --config ./balance.yaml
```
demo(): failed!💥  
  Call sequence:
    demo()

Event sequence:
Panic(1): Using assert
LogUintAddr(«this.balance», 0, @0x00a329c0648769A73afAc7F9381E08FB43dBEA72) from: 0xa329c0648769a73afac7f9381e08fb43dbea72
LogUintAddr(«sender.balance», 4294967295, IHevm@0x0000000000000000000000000000000000010000) from: 0xa329c0648769a73afac7f9381e08fb43dbea72
error BalanceTooLow 0x1 0x0

AssertionFailed(..): passing
```

**Comment by d-xo** [comment]
Thanks for the report and nice reproducer. This should be fixed once #379 is merged. Apologies for the delay.
